Sources/StreamVideoSwiftUI/Utils/Camera.swift

```diff
@@ -15,7 +15,11 @@ class Camera: NSObject, @unchecked Sendable {
     private var photoOutput: AVCapturePhotoOutput?
     private var videoOutput: AVCaptureVideoDataOutput?
     private var sessionQueue: DispatchQueue!
-    
+    private lazy var frameProcessingQueue: DispatchQueue = DispatchQueue(
+        label: "io.getstream.\(String(describing: type(of: self))).videoDataOutputQueue",
+        target: .global(qos: .userInteractive)
+    )
```

Comment:
Since the target queue is concurrent, then it feels like it still has timing issues 
Example
```
queue = DispatchQueue(label: "a", target: .global(qos: .userInteractive))
DispatchQueue.concurrentPerform(iterations: 10) { index in
  queue.async {
    print(index)
  }
}
```
prints something like
```
1
0
2
3
4
6
5
7
8
9
```
Shouldn't we just go for serial queue without a target?

Reply:
I think the problem is with your code as you are doing concurrentPerform as this isn't how the delegate will be called. The approach that i'm using is actually the one Apple recommends in their docs
```
For serial tasks, set the target of your serial queue to one of the global concurrent queues. That way, you can maintain the serialized behavior of the queue while minimizing the number of separate queues creating threads.
```